pkg/kncloudevents/event_dispatcher.go

```diff
@@ -214,7 +228,17 @@ func (d *Dispatcher) send(ctx context.Context, message binding.Message, destinat
 	}
 	additionalHeadersForDestination.Set("Prefer", "reply")
 
-	ctx, responseMessage, dispatchExecutionInfo, err := d.executeRequest(ctx, destination, message, additionalHeadersForDestination, config.retryConfig, config.oidcServiceAccount, config.transformers)
+	// Handle the event format option
+	if config.eventFormat != nil {
+		switch *config.eventFormat {
+		case Binary:
+			ctx = binding.WithForceBinary(ctx)
+		case Json:
+			ctx = binding.WithForceStructured(ctx)
+		}
+	}
+
+	ctx, responseMessage, dispatchExecutionInfo, err := d.executeRequest(ctx, destination, message, additionalHeadersForDestination, config.retryConfig, config.oidcServiceAccount, config.transformers...)
```

Comment:
nit: no need for this change, lets keep it how it is
```suggestion
	ctx, responseMessage, dispatchExecutionInfo, err := d.executeRequest(ctx, destination, message, additionalHeadersForDestination, config.retryConfig, config.oidcServiceAccount, config.transformers)
```